prov/psm2/src/psmx2_init.c

```diff
@@ -149,23 +151,37 @@ static int psmx2_read_sysfs_int(int unit, char *entry)
 
 static void psmx2_update_sep_cap(void)
 {
-	int i, n = 0;
+	int i;
+	int nctxts = 0;
+	int nfreectxts = 0;
 
-	for (i = 0; i < psmx2_env.num_devunits; i++)
-		n += psmx2_read_sysfs_int(i, "nfreectxts");
+	for (i = 0; i < psmx2_env.num_devunits; i++) {
+		nctxts += psmx2_read_sysfs_int(i, "nctxts");
+		nfreectxts += psmx2_read_sysfs_int(i, "nfreectxts");
+	}
+
+	FI_INFO(&psmx2_prov, FI_LOG_CORE,
+		"hfi1 contexts: total %d, free %d\n",
+		nctxts, nfreectxts);
 
-	FI_INFO(&psmx2_prov, FI_LOG_CORE, "Total free hfi1 contexts: %d\n", n);
+	if (nctxts > PSMX2_MAX_TRX_CTXT)
+		nctxts = PSMX2_MAX_TRX_CTXT;
 
-	/* reserve one context for regular ep */
-	if (n > 0)
-		n--;
+	if (nfreectxts > PSMX2_MAX_TRX_CTXT)
+		nfreectxts = PSMX2_MAX_TRX_CTXT;
 
-	if (n > PSMX2_MAX_TRX_CTXT)
-		n = PSMX2_MAX_TRX_CTXT;
+	psmx2_env.max_trx_ctxt = nctxts;
+	psmx2_env.sep_trx_ctxt = nfreectxts;
+
+	/*
+	 * One context is reserved for regular endpoints. It is allocated
+	 * when the domain is opened.
+	 */
+	if (!psmx2_active_fabric || !psmx2_active_fabric->active_domain)
+		--psmx2_env.sep_trx_ctxt;
```

Comment:
I think this can allow sep_trx_ctxt to go below 0.

Reply:
Thanks. Updated.